File: CSharp-Part-2/1. Arrays/Arrays/MaximalSequence/MaximalSequence.cs
// Write a program that finds the maximal sequence of equal elements in an array.

using System;

class MaximalSequence
{
    static void Main()
    {
        Console.WriteLine("Enter the contents of the array [1, 2, 3]: ");
        string inputString = Console.ReadLine();
        string[] sInputArray = inputString.Split(new string[] {", "}, StringSplitOptions.None);

        int arrSize = sInputArray.Length;
        int[] inputArray = new int[arrSize];

        bool isSequencePresent = false;
        for (int i = 0; i < arrSize; i++)
        {
            inputArray[i] = Int32.Parse(sInputArray[i]);
            if (i != 0 && inputArray[i] == inputArray[i - 1])
            {
                isSequencePresent = true;
            }
        }

        if (!isSequencePresent)
        {
            Console.WriteLine("No sequence is present in the given array.");
        }
        else 
        {
            int tempMember = 0;
            int tempCount = 0;
            int highestCount = 0;
            int arrIndex = 1;
            while (arrIndex < arrSize)
            {
                if (inputArray[arrIndex] == inputArray[arrIndex - 1])
                {
                    tempMember = inputArray[arrIndex];
                    if (tempCount > 0)
                    {
                        tempCount++;
                    }
                    else
                    {
                        tempCount += 2;
                    }
                }
                else
                {
                    tempCount = 0;
                }
                if (tempCount > highestCount)
                {
                    highestCount = tempCount;
                }
                arrIndex++;
            }
            for (int i = 0; i < highestCount; i++)
            {
                Console.Write("{0} ", tempMember);
            }
            Console.WriteLine();
        }
    }
}

File: JavaScript-Fundamentals/07. Arrays/maximal-sequence.js
// Write a script that finds the maximal sequence of equal elements in an array.

var sampleSequence = [2, 1, 1, 2, 3, 3, 2, 2, 2, 1];

function getMaximalSequence(array) {
	var currentValue = array[0],
		i,
		arrLength = array.length,
		currentCount = 1,
		maximalValueCount = 0,
		maximalElement = 0;

	for (i = 1; i < arrLength; i += 1) {
		if (currentValue === array[i]) {
			currentCount += 1;

			if (currentCount > maximalValueCount) {
				maximalValueCount = currentCount;
				maximalElement = array[i];
			}
		} else {
			if (currentCount > 1) {
				currentCount = 1;
			}

			currentValue = array[i];
		}
	}

	return Array.apply(null, new Array(maximalValueCount)).map(function(){
		return maximalElement;
	});
}

console.log(getMaximalSequence(sampleSequence));